ciphers/rot_13.cpp

```diff
@@ -0,0 +1,92 @@
+/**
+ * @brief [ROT13 Encoding and
+ * Decoding](https://en.wikipedia.org/wiki/ROT13) implementation
+ * ROT13 ("rotate by 13 places", sometimes hyphenated ROT-13) is a simple letter
+ * substitution cipher that replaces a letter with the 13th letter after it in
+ * the alphabet. ROT13 is a special case of the Caesar cipher which was
+ * developed in ancient Rome.
+ * @author [pond](https://github.com/pond-e)
+ */
+#include <iostream>  /// for IO operations
+#include <string>    /// for assert
+
+/**
+ * @namespace ciphers
+ * @brief Cipher algorithms
+ */
+namespace ciphers {
+/**
+ * @namespace rot13
+ * @brief Functions for the [Rot13 encode and
+ * decode](https://en.wikipedia.org/wiki/ROT13) implementation.
+ * */
+namespace rot13 {
+
+std::string rot13_encode(std::string s) {
+    // Encode only A~Z, a~z.
+    unsigned char tmp = 0;
```

Comment:
Let's try to use `char` only.
```suggestion
    char tmp = 0;
```